plugin-infra/go-plugin-access/src/com/thoughtworks/go/plugin/access/notification/NotificationPluginRegistry.java

```diff
@@ -47,6 +47,10 @@ public void removePluginInterests(String pluginId) {
         }
     }
 
+    public boolean isAnyPluginInterestedIn(String notificationName) {
+        return !getPluginsInterestedIn(notificationName).isEmpty();
+    }
+
     public Set<String> getPluginsInterestedIn(String notificationName) {
```

Comment:
True. You're right. I think I was thinking of ```removePluginInterests``` and ```getPluginInterests```, but it looks like ```getPluginInterests``` is used only in tests. I think ```notificationNameToPluginsInterestedMap``` should at least be marked volatile, since it is being accessed from different threads. Otherwise, when a new plugin gets registered, the other threads asking ```getPluginsInterestedIn``` might not see that change.